stdlib/Dates/test/periods.jl

```diff
@@ -175,6 +175,15 @@ end
     @test Dates.Year(2) * 3//2 == Dates.Year(3)
     @test_throws InexactError y * 0.5
     @test_throws InexactError y * 3//4
+    @test (1:1:5)*Second(5) == Second(5)*(1:1:5) == collect(1:1:5)*Second(5) == Second(5)*collect(1:1:5)
```

Comment:
```suggestion
    @test (1:1:5)*Second(5) === Second(5)*(1:1:5) === Second(5):Second(5):Second(25)
```
and get rid of the first two `isa AbstractRange` tests below.

Reply:
It would also be good to test that `(1:5)*Second(5)` gives the exact same result (including it being a `StepRange`).